packages/snaps-simulation/src/structs.ts

```diff
@@ -190,6 +190,23 @@ export const SignatureOptionsStruct = object({
   ),
 });
 
+export const NameLookupOptionsStruct = object({
+  /**
+   * The CAIP-2 chain ID. Defaults to `eip155:1`.
+   */
+  chainId: defaulted(string(), 'eip155:1'),
+
+  /**
+   * Domain name to lookup.
+   */
+  domain: optional(string()),
```

Comment:
Should we do a union? We require `address` or `domain` to be passed

Reply:
I'm not sure I see the way of doing that. Because `address` and `domain` are both top level properties and union require a bit different structure maybe? If you have an example of that, I could give it a try.

Reply:
```suggestion
export const BaseNameLookupOptionsStruct = object({
  /**
   * The CAIP-2 chain ID. Defaults to `eip155:1`.
   */
  chainId: defaulted(string(), 'eip155:1'),
});

export const NameLookupOptionsStruct = union([
  assign(BaseNameLookupOptionsStruct, object({ address: string() })),
  assign(BaseNameLookupOptionsStruct, object({ domain: string() })),
]);
```

This is one way to do it

Reply:
https://github.com/MetaMask/snaps/pull/2857/files#r1814649838

Reply:
Updated.